common/app/common/pages.cljc

```diff
@@ -350,6 +356,15 @@
 (defmethod change-spec :del-media [_]
   (s/keys :req-un [::id]))
 
+(defmethod change-spec :add-component [_]
+  (s/keys :req-un [::id ::name ::new-shapes]))
+
+(defmethod change-spec :del-component [_]
+  (s/keys :req-un [::id]))
+
+(defmethod change-spec :update-component [_]
+  (s/keys :req-un [::id ::name ::shapes]))
```

Comment:
The same here, `::shapes` is undefined spec.

Reply:
Changed. Do you think it's better to add other spec for this method, or to use the same one (it's the same data structure).

Reply:
It can be reused if is the same